### PR TITLE
Cleaned up Probabilistic

### PR DESCRIPTION
**Pull request by Salahub**
A few changes:
- updated indel() with CRISPResso data to produce results consistent with the CRISPResso distribution
- updated the time step probability to follow an exponential distribution in time rather than a uniform one
- changed the concentration probability function so that it asymptotically approaches 1, and is clearly unitless

With these changes there is one important thing in particular to note:
- the new concentration probability function does not fit the data as well as the last one, however it is unitless, will never produce an output greater than one, and still forms a reasonable approximation of the chart in Wu et al; as a result I changed the function to that and left the old, data-fitted function commented in beneath it

### CODE REVIEW

File: models/targeting/probabilistic.py [hunk 1]
```diff
@@ -49,7 +64,7 @@ def prob_cut(grna, target, concentration, dt):
           is individually important and better fit by a particular level curve
     """
     assert len(grna) == len(target)
-    prob_factor_time = dt/average_cut_time  # assume uniform distribution in time - TODO change to poisson
+    prob_factor_time = 1 - math.exp(-(average_cut_time)*dt)
```
Comment (mattsmart) [inline models/targeting/probabilistic.py]: this should be `1 - math.exp(-dt / average_cut_time)`

File: models/targeting/probabilistic.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from numpy import prod
-from random import randint
+from numpy.random import randint
+from numpy.random import choice
+import math
```
Comment (mattsmart) [inline models/targeting/probabilistic.py:1]: can we change these lines to:

```
import math
from numpy import prod
from numpy.random import choice, randint
```

File: models/targeting/probabilistic.py [hunk 3]
```diff
@@ -68,8 +83,13 @@ def nt_rand(insertion_size):
     return insertion
 
 def indel():
-    del_left = randint(0,4)
-    del_right = randint(0,4)
+    size = choice(indel_sizes, 1, p = indel_probs)
     insert = randint(0,8)
```
Comment (mattsmart) [inline models/targeting/probabilistic.py]: why is insert still random?

File: models/targeting/probabilistic.py [hunk 4]
```diff
@@ -8,6 +10,13 @@
 # average time for SpyCas9 cutting in perfect match case
 average_cut_time = 60.0
 
+# distribution of indel size from CRISPResso
```
Comment (mattsmart) [inline models/targeting/probabilistic.py:13]: delete the extra line (12), I'd like to keep the constants lumped (planning to move them later)

File: models/targeting/probabilistic.py [hunk 5]
```diff
@@ -68,8 +83,13 @@ def nt_rand(insertion_size):
     return insertion
 
 def indel():
-    del_left = randint(0,4)
-    del_right = randint(0,4)
+    size = choice(indel_sizes, 1, p = indel_probs)
     insert = randint(0,8)
-
+    total_del = abs(size - insert)[0]
+    if not total_del == 0:
```
Comment (mattsmart) [inline models/targeting/probabilistic.py]: `if total_del != 0:`

File: models/targeting/probabilistic.py [hunk 6]
```diff
@@ -68,8 +83,13 @@ def nt_rand(insertion_size):
...
+        del_left = randint(math.floor(total_del/4), total_del)
+        del_right = total_del - del_left
```
Comment (mattsmart) [inline models/targeting/probabilistic.py:103]: these lines warrant explanation in the function docstring

**Comment by mattsmart** [comment]
minor comments, can merge once you've committed the changes

**Comment by mattsmart** [comment]
also: not sure why the branch has a conflict, may need to resubmit PR..

**Comment by Salahub** [comment]
So I just fixed the conflict and added the recommended changes. Looks like my branch was just a bit behind before. Should be good now. The third commit was that issue getting resolved and the fourth was me fixing some bits of probabilistic that were messed up in the process

**Comment by mattsmart** [comment]
LGTM, merging

**Pull request merged**